### PR TITLE
ap-4508: Add class to handle linked_application_types

### PR DESCRIPTION
**Pull request by kmahern**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/AP-4508)

Describe what you did and why.

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: app/models/linked_application_type.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class LinkedApplicationType
+  LinkTypeStruct = Struct.new(:code, :description)
+
+  LINK_TYPES = [
+    { code: "FAMILY", description: "Family" },
+    { code: "LEGAL", description: "Legal" },
+  ].freeze
+
+  def self.all
+    LINK_TYPES.map { |link_type_hash| LinkTypeStruct.new(code: link_type_hash[:code], description: link_type_hash[:description]) }
```
Comment (jsugarman) [inline app/models/linked_application_type.rb:10]: you could just 

```ruby
LINK_TYPES = [
  LinkTypeStruct.new(code: "FAMILY", description: "Family"),
  LinkTypeStruct.new(code: "LEGAL", description: "Legal"),
].freeze

def self.all
  LINK_TYPES
end
```

File: spec/models/linked_application_type_spec.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+require "rails_helper"
+
+RSpec.describe LinkedApplicationType do
+  describe ".link_types" do
```
Comment (jsugarman) [inline spec/models/linked_application_type_spec.rb:4]:
```suggestion
  describe ".all" do
```

Comment (jsugarman) [reply spec/models/linked_application_type_spec.rb:4]: Possible tests to lock down the duck type and values

```ruby
it "returns a collection of Link types" do
   # expect(described_class.all).to all(be_instance_of(LinkTypeStruct))
   expect(described_class.all).to all(respond_to?(:code))
   expect(described_class.all).to all(respond_to?(:description))
end

it "returns expected collection attribute values" do
   expect(described_class.all).to match_array....
end
```

**Review by jsugarman — COMMENTED** [review]
Looks good, some simplifications and test name correction and further/different test approach suggestions

File: spec/models/linked_application_type_spec.rb (hunk 2 above)
Comment (kmahern) [reply spec/models/linked_application_type_spec.rb]: 👍 Had to tweak this suggestion slightly to get tests passing but have added tests for respond_to?(:code) and respond_to?(:description)

File: app/models/linked_application_type.rb (hunk 1 above)
Comment (kmahern) [reply app/models/linked_application_type.rb]: 👍 have made suggested change

**Review by jsugarman — APPROVED** [review]
👍

**Pull request merged**